pkg/volume/csi/csi_mounter_test.go

```diff
@@ -927,3 +936,122 @@ func TestIsCorruptedDir(t *testing.T) {
 		assert.Equal(t, test.expectedResult, isCorruptedDir, "TestCase[%d]: %s", i, test.desc)
 	}
 }
+
+func TestPodServiceAccountTokenAttrs(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIServiceAccountToken, true)()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIInlineVolume, true)()
+	scheme := runtime.NewScheme()
+	utilruntime.Must(pkgauthenticationv1.RegisterDefaults(scheme))
+	utilruntime.Must(pkgstoragev1.RegisterDefaults(scheme))
+	utilruntime.Must(pkgcorev1.RegisterDefaults(scheme))
+
+	gcp := "gcp"
+
+	tests := []struct {
+		desc              string
+		driver            *storagev1.CSIDriver
+		volumeContext     map[string]string
+		wantVolumeContext map[string]string
+	}{
+		{
+			desc: "csi driver has no ServiceAccountToken",
+			driver: &storagev1.CSIDriver{
+				ObjectMeta: meta.ObjectMeta{
+					Name: testDriver,
+				},
+				Spec: storagev1.CSIDriverSpec{},
+			},
+			wantVolumeContext: nil,
+		},
+		{
+			desc: "one token with empty string as audience",
+			driver: &storagev1.CSIDriver{
+				ObjectMeta: meta.ObjectMeta{
+					Name: testDriver,
+				},
+				Spec: storagev1.CSIDriverSpec{
+					TokenRequests: []storagev1.TokenRequest{
+						{
+							Audience: "",
+						},
+					},
+				},
+			},
+			wantVolumeContext: map[string]string{"csi.storage.k8s.io/serviceAccount.tokens": `{"":{"token":"test-ns:test-service-account:3600:[api]","expirationTimestamp":null}}`},
+		},
+		{
+			desc: "one token with non-empty string as audience",
+			driver: &storagev1.CSIDriver{
+				ObjectMeta: meta.ObjectMeta{
+					Name: testDriver,
+				},
+				Spec: storagev1.CSIDriverSpec{
+					TokenRequests: []storagev1.TokenRequest{
+						{
+							Audience: gcp,
+						},
+					},
+				},
+			},
+			wantVolumeContext: map[string]string{"csi.storage.k8s.io/serviceAccount.tokens": `{"gcp":{"token":"test-ns:test-service-account:3600:[gcp]","expirationTimestamp":null}}`},
```

Comment:
`"expirationTimestamp":null` is unexpected... shouldn't this be a string value?

Reply:
same above